webapp/src/pages/Dashboard.js

```diff
@@ -1,65 +1,147 @@
-import Header from "../components/Header";
+import api from "../api";
+import Money from "../components/Money";
+import TopNav from "../components/TopNav";
 import signOut from "../modules/signOut";
+import useAsyncFetch from "../state/useAsyncFetch";
 import { useUser } from "../state/useUser";
+import clsx from "clsx";
+import dayjs from "dayjs";
+import i18next from "i18next";
+import _ from "lodash";
 import React from "react";
+import { Alert } from "react-bootstrap";
 import Button from "react-bootstrap/Button";
 import Col from "react-bootstrap/Col";
 import Container from "react-bootstrap/Container";
+import Navbar from "react-bootstrap/Navbar";
 import Row from "react-bootstrap/Row";
+import Spinner from "react-bootstrap/Spinner";
+import Table from "react-bootstrap/Table";
 import { Link } from "react-router-dom";
 
 const Dashboard = () => {
   const { user } = useUser();
+  const { state: dashboard, loading: dashboardLoading } = useAsyncFetch(api.dashboard, {
+    default: {},
+    pickData: true,
+  });
+
   return (
-    <div className="mainContainer">
-      <Header subText="Overview" />
+    <div className="main-container">
+      <TopNav />
       <Container>
         <Row>
-          <Col className="px-3 py-4">
+          <Col className="px-3 py-2">
             {user.ongoingTrip && (
-              <Link
-                to="/map"
-                className="btn btn-sm btn-success w-100 p-2 my-2 rounded-pill"
-              >
-                You have an active ride in mobility map.
-              </Link>
+              <Alert variant="danger">
+                <p>{i18next.t("check_ongoing_trip", { ns: "dashboard" })}</p>
+                <div className="d-flex justify-content-end">
+                  <Link to="/map" className="btn btn-sm btn-danger">
+                    {i18next.t("check_ongoing_trip_button", { ns: "dashboard" })}
+                    <i
+                      className="bi bi-box-arrow-in-right mx-1"
+                      role="img"
+                      aria-label="Map Icon"
+                    ></i>
+                  </Link>
+                </div>
+              </Alert>
             )}
-            <Link
-              to="/map"
-              className="btn btn-sm btn-light w-100 p-2 my-2 text-body rounded-pill border border-primary"
-            >
-              Scooter Service
-            </Link>
-            <Link
-              to="/map"
-              className="btn btn-sm btn-light w-100 p-2 my-2 text-body rounded-pill border border-primary"
```

Comment:
If you have code that must be changed together at the same time to be correct, refactor it into something reusable (a component).